_javascripts/src/lib/search.js

```diff
@@ -0,0 +1,68 @@
+import sidebarNavigation from './sidebar_navigation'
+
+export default () => {
+  const searchBox = document.querySelector("input[type='search']")
+  const sidebarData = document.querySelectorAll('.sidebar-nav ul li')
+  const resultsPanel = document.querySelector('.results-panel')
+  const clearButton = document.querySelector('.clear-button')
+
+  const getExcerpt = (result, query) => {
+    let content = result.dataset.content
+    let searchIndex = content.search(new RegExp(query, 'i'))
+    let queryLength = query.length
+    let excerpt = content.slice(
```

Comment:
These should all be `const`. They're never reassigned, so `const` enforces that, preventing accidental changes in behaviour.